facet-reflect/src/peek/value.rs

```diff
@@ -44,7 +44,7 @@ pub struct Peek<'mem, 'facet, 'shape> {
     /// Shape of the value
     pub(crate) shape: &'shape Shape<'shape>,
 
-    invariant: PhantomData<fn(&'facet ()) -> &'facet ()>,
+    phantom: PhantomData<&'facet ()>,
```

Comment:
I don't think this is correct. If we make this lifetime covariant, then that allows `Peek::get` to return a bigger contravariant lifetime than what the facetted type contains. That is, the compiletest here will start compiling https://github.com/facet-rs/facet/pull/642/files#diff-b322a88de4968af8e4616fb2dc528ef3f69e3143a6a5190a56a7641155653d0b

Reply:
Alright, then that means we cannot use Peek in StackRunner for facet-serialize, which.. is okay I guess, we can store its components separately.